### DISCUSSION TITLE
Add Google Analytics to track our plugins

### DISCUSSION THREAD

**Discussion by andrelopez** [craftcms/console#110] [Craft Console]
In the Udemy marketplace, you as an instructor can add the Google Analytics Tracking ID. It would be great to have this feature in the Plugin Store and allow us to track our plugins at least from https://plugins.craftcms.com/ not sure if that could be possible from the CP.

> this tool will provide you with a better understanding of your visitors so you can execute your marketing efforts in an effective manner. By doing so, you are able to strategize and optimize your conversion rates and drive sales.

[Info about how Udemy handles this feature](https://blog.udemy.com/for-instructors-how-to-setup-and-use-google-analytics-to-track-your-udemy-course/)

**Comment by evanwarner** [comment]
What are you wanting to track? The stats of people who looked at a plugin page? If so I feel like this is not a high value feature, and would actually _not_ like to see GA be used for this.

I'm not sure if P&T has any custom analytics running on the plugin store but if they do perhaps there's a way to route these stats to your Craft ID dashboard. But again, I'd say this is a low priority feature.

**Comment by ryanmasuga** [comment]
If you're making money at this, anything that has to do with marketing or understanding your reach can be a high value feature. We've allowed for this at devot-ee.com (a marketplace site for another CMS) for years. Devs can see all GA info about their pages under their own GA account. It's trivial to add a second tracking ID to a page so a dev can get full stats on their own plugin pages, in their own GA account. 

At a minimum, all it would require is a single text field where the dev can put a tracking ID.  

![tracking_id](https://user-images.githubusercontent.com/988995/99327477-044f6300-2848-11eb-9b6d-891f64301635.jpg)


Then adding two lines for the developer pageview push (might be different depending on how GA is implemented in PLugin store):
```
_gaq.push(['_setAccount', 'UA-1234567-1']);
_gaq.push(['_trackPageview']);
_gaq.push(['developerTracker._setAccount','UA-2345678-1']);
_gaq.push(['developerTracker._trackPageview']);
```
